reflect: issue with exported struct members

Reported via slack.

Original repo: https://github.com/radkomih/tinygo-examples/tree/master/example1

minimized:
```
package main

import (
        "bytes"
        "reflect"
)

type U8 uint8

func (v U8) Encode(buffer *bytes.Buffer) {
        buffer.Write([]byte{0x00, 0x08})
}

type TestU8 struct {
        A U8
        B U8
        C U8
        D U8
        E U8 // fails if this additional field is added to the struct
}

//go:noinline
func encode(value interface{}, buffer *bytes.Buffer) {
        t := reflect.TypeOf(value)
        v := reflect.ValueOf(value)
        println("num field: ", v.NumField())

        for i := 0; i < v.NumField(); i++ {
                println("i=", i)
                field := v.Field(i)
                if t.Field(i).IsExported() && field.Kind() == reflect.Uint8 {
                        (field.Interface().(U8)).Encode(buffer)
                }
        }
}

func readBuffer(i int) int {
        buffer := bytes.Buffer{}
        t := TestU8{}
        println("tptr:", &t)
        encode(t, &buffer)
        return int(buffer.Bytes()[i])
}

func main() {
        v := readBuffer(1)
        println(v)
}
```

```
~/go/src/github.com/dgryski/bug/encode/tinygo-examples/example1 $ go run main.go
tptr: 0xc000108f1e
num field:  5
i= 0
i= 1
i= 2
i= 3
i= 4
8
~/go/src/github.com/dgryski/bug/encode/tinygo-examples/example1 $ tinygo run main.go
tptr: 0x000000010a655040
num field:  5
i= 0
i= 1
i= 2
i= 3
i= 4
8
~/go/src/github.com/dgryski/bug/encode/tinygo-examples/example1 $ tinygo run -target=wasi main.go
tptr: 0x000144e0
num field:  5
i= 0
panic: (reflect.Value).Interface: unexported
Error: failed to run main module `/var/folders/5b/_hr1d1fd3qn4t9vtfx5p61wm0000gp/T/tinygo1848025133/main`

Caused by:
    0: failed to invoke command default
    1: wasm trap: wasm `unreachable` instruction executed
       wasm backtrace:
           0: 0x1407 - <unknown>!runtime._panic
           1: 0x62dc - <unknown>!main.encode
           2: 0x53c2 - <unknown>!runtime.run$1
           3: 0x5106 - <unknown>!runtime.run$1$gowrapper
           4: 0x6aa2 - <unknown>!tinygo_launch
           5: 0x4faa - <unknown>!_start
       note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable to may show more debugging information

error: failed to run compiled binary /var/folders/5b/_hr1d1fd3qn4t9vtfx5p61wm0000gp/T/tinygo1848025133/main: exit status 134
```

--- a/src/reflect/value.go
+++ b/src/reflect/value.go
@@ -507,7 +507,7 @@ func (v Value) Field(i int) Value {
 		ptr := unsafe.Pointer(uintptr(v.value) + structField.Offset)
 		value := unsafe.Pointer(loadValue(ptr, fieldSize))
 		return Value{
-			flags:    0,
+			flags:    flags &^ valueFlagIndirect,
 			typecode: fieldType,
 			value:    value,
 		}

--- a/testdata/reflect.go
+++ b/testdata/reflect.go
@@ -129,6 +129,7 @@ func main() {
 		&linkedList{
 			foo: 42,
 		},
+		struct{ A, B uintptr }{2, 3},
 		// interfaces
 		[]interface{}{3, "str", -4 + 2.5i},
 	} {

--- a/testdata/reflect.txt
+++ b/testdata/reflect.txt
@@ -334,6 +334,20 @@ reflect type: ptr
     exported: false
     reflect type: int addrable=true caninterface=false
       int: 42
+reflect type: struct
+  struct: 2
+  field: 0 A
+  tag: 
+  embedded: false
+  exported: true
+  reflect type: uintptr
+    uint: 2
+  field: 1 B
+  tag: 
+  embedded: false
+  exported: true
+  reflect type: uintptr
+    uint: 3
 reflect type: slice comparable=false
   slice: interface 3 3
   pointer: true

